specification/columns/capacityreservationid.md

```diff
@@ -0,0 +1,35 @@
+# Capacity Reservation ID
+
+A Capacity Reservation ID is the identifier assigned to a [*capacity reservation*](#glossary:capacity-reservation) by the provider. Capacity Reservation ID is commonly used for scenarios to allocate charges for capacity reservation usage.
+
+The CapacityReservationId column adheres to the following requirements:
+
+* CapacityReservationId column MUST be present in the billing data when the provider supports *capacity reservations* and MUST be of type String.
+* CapacityReservationId MUST NOT contain null values when the [CapacityReservationStatus](#capacityreservationstatus) is `Unused` and SHOULD NOT contain null values when a charge is related to a capacity reservation.
```

Comment:
```suggestion
* CapacityReservationId SHOULD NOT be null when a charge is related to a capacity reservation, MUST NOT be null when a charge represents the unused portion of a *capacity reservation*, and MUST be null when a charge is not related to a *capacity reservation*.
```

The goal was to avoid a circular reference: the CapacityReservationId column MUST NOT be null when the CapacityReservationStatus is Unused, while the CapacityReservationStatus MUST be null when the CapacityReservationId is null, etc. Of course, there is still room for better phrasing :)

Reply:
Question:

Should we restrict CapacityReservationId (and consequentially CapacityReservationStatus) strictly to the ChargeCategory Usage or should we allow non-null values for Tax or possibly even Corrections as well?